stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -0,0 +1,302 @@
+//===--- TaskGroup.cpp - Task Group internal message channel ------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Object management for async child tasks that are children of a task group.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Runtime/Concurrency.h"
+#include "swift/ABI/Task.h"
+#include "swift/ABI/Metadata.h"
+#include "swift/Runtime/HeapObject.h"
+#include "TaskPrivate.h"
+
+using namespace swift;
+using GroupFragment = AsyncTask::GroupFragment;
+using FutureFragment = AsyncTask::FutureFragment;
+
+using ReadyQueueItem = GroupFragment::ReadyQueueItem;
+using ReadyStatus = GroupFragment::ReadyStatus;
+using GroupPollResult = GroupFragment::GroupPollResult;
+
+// =============================================================================
+// ==== destroy ----------------------------------------------------------------
+
+void GroupFragment::destroy() {
+  // TODO: need to release all waiters as well
+//  auto waitHead = waitQueue.load(std::memory_order_acquire);
+//  switch (waitHead.getStatus()) {
+//    case GroupFragment::WaitStatus::Waiting:
+//      assert(false && "destroying a task group that still has waiting tasks");
+//  }
```

Comment:
Not really I guess... since they never store any values, unlike in the Future case 🤔